### PR TITLE
Software Update module

### PR DESCRIPTION
**Pull request by tuxudo**
Reports back on a client's software update status

### CODE REVIEW

File: app/modules/softwareupdate/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+==============
+
+Reports on the status of software updates on the client
+
+* automaticcheckenabled - int - automatic checks enabled
+* automaticdownload - int - automatic downloads enabled
+* configdatainstall - int - automatic downloads of XProtect and GateKeeper data enabled
+* criticalupdateinstall - int - automatica downloads of critical updates enabled
+* lastattemptsystemversion - varchar(255) - last version and build of OS X that attempted an update
+* lastbackgroundccdsuccessfuldate - varchar(255) - last backgroud update seccessful date
```
Comment (bochoven) [inline app/modules/softwareupdate/README.md]: you need to replace seccessful with successful a couple of times

File: app/modules/softwareupdate/scripts/softwareupdate.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  last_xprotect_update_human_readable_time=`/bin/date -r "$last_xprotect_update_epoch_time" '+%Y-%m-%d %H:%M:%S'`
+  result="$last_xprotect_update_human_readable_time"
+  
+fi
+
+}
+
+XProtectCheck
+
+echo "$result"
```
Comment (bochoven) [inline app/modules/softwareupdate/scripts/softwareupdate.sh]: Please remove the echo

File: app/modules/softwareupdate/softwareupdate_model.php [hunk 3]
```diff
@@ -0,0 +1,146 @@
+<?php
+class Softwareupdate_model extends Model 
+{
+        public function __construct($serial='')
+        {
+                parent::__construct('id', 'softwareupdate'); //primary key, tablename
+                $this->rs['id'] = 0;
+                $this->rs['serial_number'] = $serial; $this->rt['serial_number'] = 'VARCHAR(255) UNIQUE';
+                $this->rs['automaticcheckenabled'] = 0; $this->rt['automaticcheckenabled'] = 'INT';
```
Comment (bochoven) [inline app/modules/softwareupdate/softwareupdate_model.php]: No need to specify INT, MR will do that automatically based on the initial value.

File: app/views/client/softwareupdate_tab.php [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+		}
+		else{
+			
+			// Hide
+			$('#softwareupdate-msg').text('');
+			$('#softwareupdate-view').removeClass('hide');
+			
+			// Add strings
+			if(data.automaticcheckenabled === "1" || data.automaticcheckenabled === 1) {
+				 $('#softwareupdate-automaticcheckenabled').text("Yes");
```
Comment (bochoven) [inline app/views/client/softwareupdate_tab.php]: You should localise "Yes" and "No"

Comment (tuxudo) [inline app/views/client/softwareupdate_tab.php]: I tried, but I wasn't able to find out how to make the localization framework work in JavaScript like that

Comment (bochoven) [inline app/views/client/softwareupdate_tab.php]: `i18n.t('yes')`

Comment (bochoven) [inline app/views/client/softwareupdate_tab.php]: ```i18n.t('yes')```

**Comment by bochoven** [comment]
Thanks! 

Maybe later convert dates to relative dates with moment.js

**Pull request merged**